pkgs/applications/video/ustreamer/default.nix

```diff
@@ -1,5 +1,26 @@
-{ lib, stdenv, fetchFromGitHub, libbsd, libevent, libjpeg, libdrm, pkg-config }:
-
+{ lib
+, stdenv
+, fetchFromGitHub
+, libbsd
+, libevent
+, libjpeg
+, libdrm
+, pkg-config
+, janus-gateway
+, runCommand
+, glib
+, alsa-lib
+, speex
+, jansson
+, libopus
+, withJanus ? true
+}:
+let
+  patchedJanus = runCommand "janus-gateway-ustreamer-patched" {} ''
```

Comment:
I don't think hack is necessary, might as well be achieved with some CFLAGS.

Reply:
If you have a patch, I'm more than happy to apply it. Otherwise this is fine. I'm not sure what you have in mind.

Reply:
```diff
diff --git a/pkgs/applications/video/ustreamer/default.nix b/pkgs/applications/video/ustreamer/default.nix
index 64c28cdf3ded..b612584d8b0f 100644
--- a/pkgs/applications/video/ustreamer/default.nix
+++ b/pkgs/applications/video/ustreamer/default.nix
@@ -7,7 +7,6 @@
 , libdrm
 , pkg-config
 , janus-gateway
-, runCommand
 , glib
 , alsa-lib
 , speex
@@ -15,12 +14,6 @@
 , libopus
 , withJanus ? true
 }:
-let
-  patchedJanus = runCommand "janus-gateway-ustreamer-patched" {} ''
-    cp -r --no-preserve=mode ${janus-gateway.dev} $out
-    substituteInPlace $out/include/janus/plugins/plugin.h --replace 'refcount.h' '../refcount.h'
-  '';
-in
 stdenv.mkDerivation rec {
   pname = "ustreamer";
   version = "6.12";
@@ -38,7 +31,7 @@ stdenv.mkDerivation rec {
     libjpeg
     libdrm
   ] ++ lib.optionals withJanus [
-    patchedJanus
+    janus-gateway
     glib
     alsa-lib
     jansson
@@ -53,6 +46,9 @@ stdenv.mkDerivation rec {
     "WITH_V4P=1"
   ] ++ lib.optionals withJanus [
     "WITH_JANUS=1"
+    # Workaround issues with Janus C Headers
+    # https://github.com/pikvm/ustreamer/blob/793f24c4/docs/h264.md#fixing-janus-c-headers
+    "CFLAGS=-I${lib.getDev janus-gateway}/include/janus"
   ];
 
   enableParallelBuilding = true;
```

Reply:
Excellent, thank you so much!